test/libponyc/verify.cc

```diff
@@ -5869,3 +5869,41 @@ TEST_F(VerifyTest,
 
   TEST_COMPILE(src);
 }
+
+// From issue #3765
+TEST_F(VerifyTest, NonRootStructBehaviourParam)
+{
+  const char* src =
+    "struct iso AA\n"
+
+    "actor Main\n"
+      "new create(env: Env) =>\n"
+        "inspect(AA)\n"
+
+      "be inspect(a: AA val) => None";
+
+  TEST_ERRORS_1(src,
+    "cannot use a struct type as parameter to a behaviour; "
```

Comment:
```suggestion
    "cannot use struct as parameter to a behaviour; "
```

Reply:
i'm not thrilled with this change, but its better. you can use a struct.

maybe:

"only \"tag"\ structs can be used as a parameter to a behavior; "
"you'll need to change the capability or wrap the struct in a class or actor"

something kind of like that?